js/models/messages.js

```diff
@@ -1900,33 +1912,63 @@
       return message;
     },
 
-    async handleDataMessage(initialMessage, confirm) {
-      // This function is called from the background script in a few scenarios:
-      //   1. on an incoming message
-      //   2. on a sent message sync'd from another device
-      //   3. in rare cases, an incoming message can be retried, though it will
-      //      still go through one of the previous two codepaths
-      const ourNumber = textsecure.storage.user.getNumber();
-      const message = this;
-      const source = message.get('source');
-      const type = message.get('type');
-      let conversationId = message.get('conversationId');
-      const authorisation = await libloki.storage.getGrantAuthorisationForSecondaryPubKey(
-        source
+    async handleSecondaryDeviceFriendRequest(pubKey) {
```

Comment:
Moved from `message_receiver.js`.
returns true if and only if a pubkey is the pubkey of a secondary device and we are already friends with the primary device of that user